driver/conv_driver.hpp

```diff
@@ -201,16 +210,197 @@ bool readBufferFromFile(T* data, size_t dataNumItems, const char* fileName)
     }
 }
 
+template <typename Tgpu>
+class GpumemTensor
+{
+    std::unique_ptr<GPUMem> dev;
+    tensor<Tgpu> host;
+    bool is_gpualloc = false;
+
+public:
+    void SetGpuallocMode(bool v) { is_gpualloc = v; }
+    tensor<Tgpu>& GetTensor() { return host; }
+
+    void AllocOnHost(miopenTensorDescriptor_t t)
+    {
+        host = tensor<Tgpu>(miopen::deref(t));
+        if(is_gpualloc) // We do not need host data.
+        {
+            host.data.clear();
+            host.data.shrink_to_fit(); // To free host memory.
+        }
+    }
+
+    std::vector<Tgpu>& GetVector()
+    {
+        if(is_gpualloc)
+            MIOPEN_THROW("[MIOpenDriver] GpumemTensor::GetVector should not be called in "
+                         "'--gpualloc 1' mode");
+        return host.data;
+    }
+
+    Tgpu* GetVectorData() { return is_gpualloc ? nullptr : host.data.data(); }
+    std::size_t GetVectorSize() const { return is_gpualloc ? 0 : host.data.size(); }
+
+    void
+    InitHostData(const size_t sz,     //
+                 const bool do_write, // If set to false, then only generate random data. This is
+                                      // necessary to reproduce values in input buffers even if some
+                                      // directions are skipped. For example, inputs for Backward
+                                      // will be the same for both "-F 0" and "-F 2".
+                 std::function<Tgpu()> generator)
+    {
+        if(is_gpualloc)
+        {
+            /// In gpualloc mode, we do not care about reproducibility of results, because
+            /// validation is not used. Therefore, we do not have to always generate random value
+            /// (\ref move_rand)
+            return;
+        }
+
+        for(int i = 0; i < sz; ++i)
+        {
+            /// \anchor move_rand
+            /// Generate random value, even if buffer is unused. This provides the same
+            /// initialization of input buffers regardless of which kinds of
+            /// convolutions are currently selectedfor testing (see the "-F" option).
+            /// Verification cache would be broken otherwise.
+            auto val = generator();
+            if(do_write)
+                GetVector()[i] = val;
+        }
+    }
+
+    status_t AllocOnDevice(stream, context_t ctx, const size_t sz)
+    {
+        dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, sz, sizeof(Tgpu)));
```

Comment:
It's not critical, but consider useing `std::make_unique`
```suggestion
        dev = std::make_unique<GPUMem>(ctx, sz, sizeof(Tgpu));
```
Also if a reallocation is required, it would be better to explicitly free the buffer before the new allocation, because in this particular case it requires `(oldSz + sz) * sizeof(Tgpu)` amount of memory instead of just a `sz`. May or may not be critical.
```suggestion
        dev.reset();
        dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, sz, sizeof(Tgpu))); //or std::make_unique()
```

Reply:
@CAHEK7 Reallocation is not required at all yet. By design, each buffer is allocated only once at the beginning and then freed once at the end. Thanks for this comment anyway.

Reply:
[Gentle reminder] @CAHEK7 Please mark this review thread as "resolved" or provide some other kind of feedback.

Reply:
What do you think about `make_unique`?

Reply:
@CAHEK7 Done in c28d88037